### ISSUE TITLE
Stuck on splash screen at launch

### ISSUE THREAD

**Issue by rschroll** [jupyterlab/jupyterlab_app#108]
On launching the app with `npm run build:all`, the splash screen opens and animates, but nothing progresses from there.  Stdout from the process is below.

This is on Ubuntu Linux 14.04, with node v6.11.2 and npm 3.10.10.  I'm currently at 85cf025.

-----
```
> jupyterlab_app@0.1.0 build:all /home/rschroll/jupyterlab_app
> npm run build && npm start


> jupyterlab_app@0.1.0 build /home/rschroll/jupyterlab_app
> npm run build:main && npm run build:browser


> jupyterlab_app@0.1.0 build:main /home/rschroll/jupyterlab_app
> webpack --config webpack.main.js


[at-loader] Using typescript@2.3.4 from typescript and "tsconfig.json" from /home/rschroll/jupyterlab_app/tsconfig.json.


[at-loader] Checking started in a separate process...

[at-loader] Ok, 0.89 sec.
Hash: ea29979e9db284a5ec8f
Version: webpack 2.7.0
Time: 3919ms
             Asset       Size  Chunks             Chunk Names
        index.html  985 bytes          [emitted]  
    main.bundle.js     138 kB       0  [emitted]  main
main.bundle.js.map     193 kB       0  [emitted]  main
   [0] ./~/@phosphor/algorithm/lib/iter.js 5.61 kB {0} [built]
   [2] ./src/ipc.ts 3.79 kB {0} [built]
   [3] ./~/@phosphor/algorithm/lib/index.js 1.03 kB {0} [built]
   [4] ./src/main/app.ts 7.53 kB {0} [built]
  [12] ./~/@phosphor/algorithm/lib/range.js 3.33 kB {0} [built]
  [13] ./~/@phosphor/algorithm/lib/reduce.js 1.99 kB {0} [built]
  [14] ./~/@phosphor/algorithm/lib/repeat.js 2.36 kB {0} [built]
  [15] ./~/@phosphor/algorithm/lib/retro.js 2.19 kB {0} [built]
  [16] ./~/@phosphor/algorithm/lib/sort.js 2.15 kB {0} [built]
  [21] ./src/main/main.ts 412 bytes {0} [built]
  [22] ./src/main/menu.ts 3.78 kB {0} [built]
  [23] ./src/main/server.ts 7.96 kB {0} [built]
  [24] ./src/main/shortcuts.ts 4.28 kB {0} [built]
  [25] ./src/main/state.ts 9.04 kB {0} [built]
  [26] ./src/main/window.ts 4.88 kB {0} [built]
    + 17 hidden modules

> jupyterlab_app@0.1.0 build:browser /home/rschroll/jupyterlab_app
> webpack --config webpack.browser.js


[at-loader] Using typescript@2.3.4 from typescript and "tsconfig.json" from /home/rschroll/jupyterlab_app/tsconfig.json.


[at-loader] Checking started in a separate process...

[at-loader] Ok, 1.024 sec.
Hash: cfa4587654efecc5cea0
Version: webpack 2.7.0
Time: 13846ms
                                 Asset       Size  Chunks                    Chunk Names
  42bbe4eefcde1297b11dc4b6491e9746.ttf     174 kB          [emitted]         
  3275372707f7d039afebb96123631395.png    5.44 kB          [emitted]         
  995a1d96b1fef026c61ac03f4c6ff841.png  210 bytes          [emitted]         
  200b3fc810ff360a023c4cc309f77e5a.png  217 bytes          [emitted]         
  1c8b268ec020a3aa5cd211cdda05fe2b.png  214 bytes          [emitted]         
  674f50d287a8c48dc19ba404d20fe713.eot     166 kB          [emitted]         
  42755f5d26c8ce544c6c5ae41fe2135e.map    3.17 kB          [emitted]         
  30e4639d54822ef7e3f80f11a73ebc3f.map    1.17 kB          [emitted]         
af7ae505a9eed503f8b8e6982036873e.woff2    77.2 kB          [emitted]         
 fee66e712a8a08eef5805a46892932ad.woff      98 kB          [emitted]         
  b06871f281fee6b241d60582ae9369b9.ttf     166 kB          [emitted]         
  ee7b96fa85d8fdb8c126409326ac2d2b.ttf     171 kB          [emitted]         
  1eb7a893589ddce89d81cdb22a356660.ttf     175 kB          [emitted]         
  7a564d5604efdab6ca2325dcced48457.png  210 bytes          [emitted]         
  3e1af3ef546b9e6ecef9f3ba197bf7d2.ttf     172 kB          [emitted]         
  a3c3082872743102af82207d10076209.ttf     212 kB          [emitted]         
  d77c7593ff187c79a73360a5edd1da75.ttf     174 kB          [emitted]         
  4f22015ae8668226a7f7e8413fe93a81.ttf     180 kB          [emitted]         
  1066e54d79f902a2ff4d864c9a38a183.ttf     213 kB          [emitted]         
  fcad2f97acb3c868bd1ca7629d36d80e.svg    16.9 kB          [emitted]         
  912ec66d7572ff821749319396470bde.svg     444 kB          [emitted]  [big]  
  2e6d2674cbb01a89b6044ae1dd256066.svg    24.1 kB          [emitted]         
                   0.browser.bundle.js     892 kB       0  [emitted]  [big]  
                     browser.bundle.js    7.85 MB       1  [emitted]  [big]  main
               0.browser.bundle.js.map    1.08 MB       0  [emitted]         
                 browser.bundle.js.map    9.39 MB       1  [emitted]         main
  [16] ./~/style-loader/addStyles.js 6.91 kB {1} [built]
  [41] (webpack)/buildin/global.js 509 bytes {1} [built]
  [67] ./~/react/react.js 56 bytes {1} [built]
 [219] ./src/browser/app.tsx 9.15 kB {1} [built]
 [220] ./~/react-dom/index.js 59 bytes {1} [built]
 [285] ./src/browser/extensions/electron-extension/index.ts 5.79 kB {1} [built]
 [539] ./~/@jupyterlab/theming/style/index.css 871 bytes {1} [built]
 [540] ./~/font-awesome/css/font-awesome.min.css 892 bytes {1} [built]
 [541] ./src/browser/style/main.css 920 bytes {1} [built]
 [759] ./src/browser/extensions/index.ts 1.42 kB {1} [built]
 [762] ./src/browser/index.tsx 639 bytes {1} [built]
 [831] ./~/css-loader!./~/@jupyterlab/theming/style/index.css 3.85 kB {1} [built]
 [861] ./~/css-loader!./~/font-awesome/css/font-awesome.min.css 33.3 kB {1} [built]
 [870] ./~/css-loader!./src/browser/style/main.css 3.12 kB {1} [built]
[1010] ./~/react-dom/lib/ReactDOM.js 5.17 kB {1} [built]
    + 1416 hidden modules

WARNING in ./~/xterm/lib/addons/search/search.js
10:31-45 Critical dependency: the request of a dependency is an expression

> jupyterlab_app@0.1.0 start /home/rschroll/jupyterlab_app
> electron .
```

**Comment by lucbouchard1** [comment]
Hi @rschroll. It's hard for me to say for sure, but I think this has something to do with the Jupyter server not starting up properly as a subprocess. Since the application doesn't ship with a Jupyter server, it needs to be able to launch one that is already installed. We're actively working on improving that aspect of the application, but it's been tricky given everyone has different python envs, PATHS, etc.

Couple questions: 
- Is the Jupyter server in your PATH? 
- What version of the Jupyter server are you using.

**Comment by rschroll** [comment]
I have conda set up to be my default Python.  It's installed in my home directory with my $PATH being modified in my `~/.bashrc`.  From my command line:
```
rschroll@mando:~$ which jupyter
/home/rschroll/miniconda2/bin/jupyter
rschroll@mando:~$ jupyter --version
4.3.0
```
Is there anyway to get the output of the subprocess, so I can examine it?

FWIW, I've always managed to launch the jupyter server without any problem in Juno.  The relevant code is here: https://github.com/rschroll/juno/blob/master/src/notebook.js#L93

**Comment by lucbouchard1** [comment]
There no code to examine the output of the Jupyter server right now, but it's something we should add. I created an issue, #110, to talk about it.

All the server spawning code is in [server.ts](https://github.com/jupyterlab/jupyterlab_app/blob/e97fd7d04ce2ff9aad348740264953f1138962b0/src/main/server.ts). That would be the place to start debugging this.

**Comment by rschroll** [comment]
After digging around a little, I found that, while I had version 4.3.0 of jupyter, I had version 4.2.0 of jupyter notebook.  Apparently, JupyterLab recently moved to requiring version 4.3 of the notebook.

After a bit of conda confusion, I've upgraded and now have jupyter 4.1.0 and jupyter notebook 4.3.1.  I really don't understand how that happened.  But now the local app works.

I'll close this bug, but better error reporting is a must.
